### PR TITLE
milestone #3 

### PR DESCRIPTION
**Pull request by blackpintz**
##  Tic-tac-toe game logic

Encapsulated different parts of the game logic in classes.

### CODE REVIEW

**Comment by blackpintz** [comment]
Made the requested changes.

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
+require 'rubygems'
+require 'terminal-table/import'
+
+$num_arr = []
+$playing_numbers = (1..9).to_a
+$player_status = 1
+$winner = false
```
Comment (maelfosso) [inline lib/game.rb:7]: Global variables are not allowed.
Use **instance variable**

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    $num_arr.push(@num_played)
+  end
+end
+
+class PlayerTwo < PlayerOne
+  def initialize(num_played)
+    super
+    $player_status = 1
+  end
+end
```
Comment (maelfosso) [inline lib/game.rb:26]: Can be done with a class **Player** and you instantiate **player_one** and **player_two** as object

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+class Board
+  attr_accessor :arry_one, :arry_two, :arry_three
```
Comment (maelfosso) [inline lib/game.rb:29]: It will be better to use an instance variable **array** as an array of size 9 or size 3 * 3

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      i += 1
+      break if check == true
+    end
+    check == true ? $winner = true : nil
+  end
+
+  def linear_check
+    check = (@arry_one.all?(0) || @arry_one.all?('X')) || (@arry_two.all?(0) || @arry_two.all?('X')) || (@arry_three.all?(0) || @arry_three.all?('X'))
+    check == true ? $winner = true : nil
+  end
```
Comment (maelfosso) [inline lib/game.rb:73]: All these 3 methods can be methods inside the **Board** class

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        $playing_numbers.map! do |value|
+          value == val ? value = 0 : value
+        end
+      end
+      $board_number = $playing_numbers.each_slice(3).to_a
+      my_board = Board.new($board_number[0], $board_number[1], $board_number[2])
+      check_win = CheckWinner.new($board_number[0], $board_number[1], $board_number[2])
+      check_win.vertical_check
+      check_win.diagonal_check
+      check_win.linear_check
```
Comment (maelfosso) [inline lib/game.rb:94]: Define a **win** method inside the **Board** class that will call those 3 methods and returns **true** if any of them is returning **true**

File: lib/game.rb [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+  def draw_board
+    board = table do |t|
+      t << @arry_one
+      t.add_separator
+      t << @arry_two
+      t.add_separator
+      t << @arry_three
+    end
+    board
```
Comment (maelfosso) [inline lib/game.rb:44]: Re-think that method according to the indication gives up there

**Comment by blackpintz** [comment]
Removed global variables.
Defined classes - Board, Player and GameLogic.
Used an instance variable in the Board class.

**Comment by Anna-Myzukina** [comment]
# Hi team! :clap: 
### Good job  in Milestone 3 :eyes: 

### Just few changes before it can be approved :100: 
##### Implementation details

* [ ] Use attr_readers/attr_writers instead of attr_accessors. The student must be aware that by using it they allow writing that property from outside the class. That variable can be set to anything. <TODO: explain why> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

## [OPTIONAL]

* [x] Using `require_relative` for files in your directory instead of `require` is a ruby convention.


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by OdongAlican** [comment]
Replaced attr_accessors with attr_readers

**Pull request merged**